doc/src/manual/memory-management.md

```diff
@@ -0,0 +1,203 @@
+
+Julia uses automatic memory management through its built-in garbage collector (GC). This section provides an overview of how Julia manages memory and how you can configure and optimize memory usage for your applications.
+
+## [Garbage Collection Overview](@id man-gc-overview)
+
+Julia features a garbage collector with the following characteristics:
+
+* **Non-moving**: Objects are not relocated in memory during garbage collection
+* **Generational**: Younger objects are collected more frequently than older ones
+* **Parallel and partially concurrent**: The GC can use multiple threads and run concurrently with your program
+* **Mostly precise**: The GC accurately identifies object references with minimal conservative scanning
+
+The garbage collector automatically reclaims memory used by objects that are no longer reachable from your program, freeing you from manual memory management in most cases.
+
+## [Memory Architecture](@id man-memory-architecture)
+
+Julia uses a two-tier allocation strategy:
+
+* **Small objects** (≤ 2032 bytes): Allocated using a fast per-thread pool allocator
+* **Large objects** (> 2032 bytes): Allocated directly through the system's `malloc`
+
+This hybrid approach optimizes for both allocation speed and memory efficiency, with the pool allocator providing fast allocation for the many small objects typical in Julia programs.
+
+## [System Memory Requirements](@id man-system-memory)
+
+### Swap Space
+
+Julia's garbage collector is designed with the expectation that your system has adequate swap space configured. The GC uses heuristics that assume it can allocate memory beyond physical RAM when needed, relying on the operating system's virtual memory management.
+
+If your system has limited or no swap space, you may experience out-of-memory errors during garbage collection. In such cases, you can use the `--heap-size-hint` option to limit Julia's memory usage.
+
+### Memory Hints
+
+You can provide a hint to Julia about the maximum amount of memory to use:
+
+```bash
+julia --heap-size-hint=4G  # To set the hint to ~4GB
+julia --heap-size-hint=50% # or to 50% of physical memory
+```
+
+The `--heap-size-hint` option tells the garbage collector to trigger collection more aggressively when approaching the specified limit. This is particularly useful in:
+
+* Containers with memory limits
+* Systems without swap space
+* Shared systems where you want to limit Julia's memory footprint
+
+You can also set this via the `JULIA_HEAP_SIZE_HINT` environment variable:
+
+```bash
+export JULIA_HEAP_SIZE_HINT=2G
+julia
+```
+
+## [Multithreaded Garbage Collection](@id man-gc-multithreading)
+
+Julia's garbage collector can leverage multiple threads to improve performance on multi-core systems.
+
+### GC Thread Configuration
+
+By default, Julia uses multiple threads for garbage collection:
+
+* **Mark threads**: Used during the mark phase to trace object references (default: same as compute threads (thread 1 is shared))
```

Comment:
```suggestion
* **Mark threads**: Used during the mark phase to trace object references (default: 1, which is shared with the compute thread if there is only one, otherwise half the number of compute threads)
```

the nested parenthetical here reads a bit odd, and I'm not sure how to square the information with https://docs.julialang.org/en/v1/manual/multi-threading/#Multiple-GC-Threads stating that #mark threads = 1/2 #compute threads

Reply:
Maybe the point I was trying to highlight isn't important, but the first gc mark thread is the same thread as thread 1.
```
% ./julia -t6 -q
julia> versioninfo()
...
Threads: 6 default, 1 interactive, 6 GC (on 6 virtual cores)

julia> Threads.maxthreadid()
12
```
(not 13)

I've been confused by that before. But maybe that's too niche.

Reply:
now I'm confused even more 😁, if there are 6 worker threads and 6 GC threads, how is that "half the number of compute threads" ? are the original docs wrong?

Reply:
Yes, the default number of GC threads is now --threads.

Reply:
https://github.com/JuliaLang/julia/pull/53608

Reply:
Ok. I added a docs fix, not not sure if I found every instance. It's a hard thing to search for.